include/types.hpp

```diff
@@ -92,5 +94,16 @@ using HWVerList = std::vector<std::pair<std::string, std::string>>;
 using SystemTypeMap =
     std::unordered_map<std::string, std::pair<std::string, HWVerList>>;
 
+using Path = std::string;
+using Record = std::string;
+using Keyword = std::string;
+
+using IpzType = std::tuple<Record, Keyword, BinaryVector>;
```

Comment:
Why we can't use std::string in place of BinaryVector here?

Reply:
to take control characters as input we can't use std::string. we rely on hex values.